ceph/ceph.py

```diff
@@ -881,7 +882,7 @@ def generate_repository_file(base_url, repos):
         """
         repo_file = ""
         for repo in repos:
-            repo_to_use = base_url + "compose/" + repo + "/x86_64/os/"
+            repo_to_use = os.path.join(base_url, "compose", repo, "x86_64/os/")
```

Comment:
It is kind of misleading though it achieves the requirement, IMO
```
repo = repo.rstrip('/')
repo_to_use = f"{base_url}compose/{repo}/x86_64/os"
```

Reply:
os.path.join resolves `/` or `//`

Reply:
@psathyan is it okay if we avoid repo = repo.rstrip('/') and allow it to add two '/' as it wouldn't make any issues ?

Reply:
I think `os.path` is a good solution. Usually for URLs I use `posixpath.join()` though, because that will use forward slashes everywhere. Not like we're going to run on MS Windows though :)

Reply:
(also, reading this code, we should probably rename `repo_to_use` to `baseurl` in a separate PR)

Reply:
> @psathyan is it okay if we avoid repo = repo.rstrip('/') and allow it to add two '/' as it wouldn't make any issues ?

@VasishtaShastry let's keep it clean. We should strip or add the `/`... 

The double slash is a problem unless the gateway is capable of redirecting the request.